packages/hardhat-core/src/internal/hardhat-network/provider/node.ts

```diff
@@ -2342,7 +2364,7 @@ Hardhat Network's forking functionality only works with blocks from at least spu
   private async _persistIrregularWorldState(): Promise<void> {
     this._irregularStatesByBlockNumber.set(
       this.getLatestBlockNumber(),
-      await this._vm.makeSnapshot()
```

Comment:
This breaks one of the forking tests: https://github.com/NomicFoundation/hardhat/actions/runs/4956863372/jobs/8867848141?pr=3916#step:9:5021